docs/conf.py

```diff
@@ -59,45 +58,68 @@
+intersphinx_mapping['sphinx_automodapi'] = ('https://sphinx-automodapi.readthedocs.io/en/stable/', None)  # noqa: F405, E501
+intersphinx_mapping['packagetemplate'] = ('https://docs.astropy.org/projects/package-template/en/latest/', None)  # noqa: F405, E501
+intersphinx_mapping['h5py'] = ('http://docs.h5py.org/en/stable/', None)  # noqa: F405
 
 # List of patterns, relative to source directory, that match files and
 # directories to ignore when looking for source files.
-exclude_patterns.append('_templates')
-exclude_patterns.append('_pkgtemplate.rst')
-exclude_patterns.append('**/*.inc.rst')  # .inc.rst mean *include* files, don't have sphinx process them
+exclude_patterns.append('_templates')  # noqa: F405
+exclude_patterns.append('_pkgtemplate.rst')  # noqa: F405
+exclude_patterns.append('**/*.inc.rst')  # .inc.rst mean *include* files, don't have sphinx process them  # noqa: F405, E501
 
 # Add any paths that contain templates here, relative to this directory.
 if 'templates_path' not in locals():  # in case parent conf.py defines it
     templates_path = []
 templates_path.append('_templates')
 
+# Grab minversion from setup.cfg
+setup_cfg = configparser.ConfigParser()
+setup_cfg.read(os.path.join(os.path.pardir, 'setup.cfg'))
+__minimum_python_version__ = setup_cfg['options']['python_requires'].replace('>=', '')
+project = u'Astropy'
+astropy_dist = get_distribution(project)
+astropy_req = astropy_dist.requires(extras=('all', ))
+for cur_req in astropy_req:
+    if cur_req.name == 'numpy':
+        __minimum_numpy_version__ = cur_req.specs[0][1]
+    elif cur_req.name == 'pyerfa':
+        __minimum_erfa_version__ = cur_req.specs[0][1]
+    elif cur_req.name == 'scipy':
+        __minimum_scipy_version__ = cur_req.specs[0][1]
+    elif cur_req.name == 'PyYAML':
+        __minimum_yaml_version__ = cur_req.specs[0][1]
+    elif cur_req.name == 'asdf':
+        __minimum_asdf_version__ = cur_req.specs[0][1]
+    elif cur_req.name == 'matplotlib':
+        __minimum_matplotlib_version__ = cur_req.specs[0][1]
+    elif cur_req.name == 'ipython':
+        __minimum_ipython_version__ = cur_req.specs[0][1]
+
 # This is added to the end of RST files - a good place to put substitutions to
 # be used globally.
 rst_epilog += f"""
-.. |minimum_python_version| replace:: {astropy.__minimum_python_version__}
-.. |minimum_numpy_version| replace:: {astropy.__minimum_numpy_version__}
-.. |minimum_erfa_version| replace:: {astropy.__minimum_erfa_version__}
-.. |minimum_scipy_version| replace:: {astropy.__minimum_scipy_version__}
-.. |minimum_yaml_version| replace:: {astropy.__minimum_yaml_version__}
-.. |minimum_asdf_version| replace:: {astropy.__minimum_asdf_version__}
-.. |minimum_matplotlib_version| replace:: {astropy.__minimum_matplotlib_version__}
-.. |minimum_ipython_version| replace:: {astropy.__minimum_ipython_version__}
+.. |minimum_python_version| replace:: {__minimum_python_version__}
```

Comment:
Super-fancy might be something like
```
rst_epilog += "\n".join(
    f".. |minimum_{name}_version| replace:: {minvers(name)}"
    for name in ('python', 'numpy', ...))
```
where you define a quick `minvers` function that does the look-up, like
```
def minvers(name):
    if name == 'yaml':
        name = 'PyYaml'
    for cur_req in astropy_req:
       if cur_req.name == name:
           return cur_req.specs[0][1]
    raise ValueError(f"{name} not among requirements!")
```

Reply:
Fancy!